openedx/core/djangoapps/user_authn/views/tests/test_filters.py

```diff
@@ -0,0 +1,208 @@
+
+    def setUp(self):  # pylint: disable=arguments-differ
+        super().setUp()
+        self.url = reverse("user_api_registration")
+        self.user_info = {
+            "email": "user@example.com",
+            "name": "Test User",
+            "username": "test",
+            "password": "password",
+            "honor_code": "true",
+        }
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.student.registration.requested.v1": {
+                "pipeline": [
+                    "openedx.core.djangoapps.user_authn.views.tests.test_filters.TestRegisterPipelineStep",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_register_filter_executed(self):
+        """
+        Test whether the student register filter is triggered before the user's
+        registration process.
+
+        Expected result:
+            - PreRegisterFilter is triggered and executes TestRegisterPipelineStep.
+            - The user's username is updated.
+        """
+        self.client.post(self.url, self.user_info)
+
+        user = User.objects.filter(username=f"{self.user_info.get('username')}-OpenEdx")
+        self.assertTrue(user)
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.student.registration.requested.v1": {
+                "pipeline": [
+                    "openedx.core.djangoapps.user_authn.views.tests.test_filters.TestStopRegisterPipelineStep",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_register_filter_prevent_registration(self):
+        """
+        Test prevent the user's registration through a pipeline step.
+
+        Expected result:
+            - PreRegisterFilter is triggered and executes TestStopRegisterPipelineStep.
+            - The user's registration stops.
+        """
+        response = self.client.post(self.url, self.user_info)
+
+        self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
+
+
+@skip_unless_lms
+class LoginFiltersTest(UserAPITestCase):
+    """
+    Tests for the Open edX Filters associated with the user login process.
+
+    This class guarantees that the following filters are triggered during the user's login:
+
+    - PreLoginFilter
+    """
+
+    def setUp(self):  # pylint: disable=arguments-differ
+        super().setUp()
+        self.user = UserFactory.create(
+            username="test",
+            email="test@example.com",
+            password="password",
+        )
+        self.user_profile = UserProfileFactory.create(user=self.user, name="Test Example")
+        self.url = reverse('login_api')
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.student.login.requested.v1": {
+                "pipeline": [
+                    "openedx.core.djangoapps.user_authn.views.tests.test_filters.TestLoginPipelineStep",
```

Comment:
Does it do the right thing when this configuration is specified but the pipeline is empty?

Reply:
Also, should we have a test with more than one step?

Reply:
1. Yes, it behaves like a noop. I'll attach some tests:
- Unit-test from openedx-filters: [test_run_empty_pipeline](https://github.com/eduNEXT/openedx-filters/blob/a522220cae5f6ec503b9e28f0514d0ea04cde258/openedx_filters/tests/test_tooling.py#L262)
- On my dev environment: 
Using this configuration ->
```
OPEN_EDX_FILTERS_CONFIG = {
    "org.openedx.learning.student.login.requested.v1": {
        "fail_silently": False,
        "pipeline": [],
        "log_level": "debug",
    },
}
```
![gif-noop-filter](https://user-images.githubusercontent.com/64440265/145471123-6c4be00e-2746-4718-8b61-e5957f1e5272.gif)